### ISSUE TITLE
Different updating behavior between modes

### ISSUE THREAD

**Issue by KingSora** [nuxt/nuxt.js#7704]
### Versions

- nuxt: v2.13.2

### Reproduction
Codesandbox: https://codesandbox.io/s/elegant-dhawan-z7vlo?file=/components/Test.vue
Please open Additional Details for the full explanation.

<details open>
<summary>Additional Details</summary>

I have a vue component which uses a vanilla js plugin in order to modify the DOM of the vue component:
```vue
<template>
  <div class="test-host">test-host
    <div class="test-child">test-child
      <div class="test-child-child">test-child-child
        <div>
          <slot></slot>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="js">
const plugin = (target) => {
  if(target) {
    target.classList.add('test-host-plugin');
    const child = target.querySelector('.test-child');
    const childChild = target.querySelector('.test-child-child');

    child.classList.add('test-child-plugin');
    childChild.classList.add('test-child-child-plugin');
    
    const hostElm = document.createElement('div');
    const childElm = document.createElement('div');
    const childChildElm = document.createElement('div');

    hostElm.innerHTML  = "host plugin item";
    childElm.innerHTML  = "child plugin item";
    childChildElm.innerHTML  = "child child plugin item";

    target.append(hostElm);
    child.append(childElm);
    childChild.append(childChildElm);
  }
};

export default {
  name: 'test-test',

  data() {
      return {};
  },

  updated() {
  },

  mounted() {
    plugin(this.$el);
  },
}
</script>

<style scoped>
.test-host-plugin {
  color: red;
}
.test-child-plugin {
  color: blue;
}
.test-child-child-plugin {
  color: green;
}
</style>
```

After I click on a button which forces this component to re-render, the `classes` of the `test-child` and `test-child-child` elements gets resetted for some reason (maybe because the were modified by the vanilla js plugin?). 

This doesn't happen at all in the `spa` mode. 

</details>

### Steps to reproduce
In the Sandbox, click on the "TOGGLE" button. The colored text turns red because the class names changed.

### What is Expected?
The class names shouldn't be modified at all.
Basically same behavior as in `spa` mode.

### What is actually happening?
The class names are modified.
